applications/nrf5340_audio/src/bluetooth/bt_rendering_and_capture/volume/bt_vol_ctlr.c

```diff
@@ -70,49 +98,149 @@ static bool vcs_client_peer_exists(struct bt_conn *conn)
 
+	for (size_t i = 0; i < ARRAY_SIZE(vcs_client_peer); i++) {
+		LOG_DBG("Saving vol %u for dev %u", volume_out, i);
+		vcs_client_peer[i].master_vol = volume_out;
+		vcs_client_peer[i].unknown_vol = false;
+
+		if (!vcs_peer_alive(vcs_client_peer[i].peer)) {
+			LOG_DBG("Peer %u is not alive", i);
+			continue;
+		}
+
+		if (vcs == vcs_client_peer[i].peer && !include_caller) {
+			LOG_DBG("Caller not included");
+			continue;
+		}
+
+		ret = bt_vcp_vol_ctlr_set_vol(vcs_client_peer[i].peer, volume_out);
+
+		if (ret) {
+			LOG_ERR("Vol sync failed to remote device %u, err = %d", i, ret);
+		}
+	}
+}
+
+static void vol_set_cb_handler(struct bt_vcp_vol_ctlr *vol_ctlr, int err)
+{
+	int idx;
+
+	if (err) {
+		LOG_WRN("VCS vol set callback error: %d", err);
+		return;
+	}
+
+	idx = vcs_peer_index_get(vol_ctlr);
+	if (idx < 0) {
+		LOG_ERR("VCS peer not found");
+		return;
+	}
+
+	LOG_DBG("Saving vol %u for dev %u", vol_ctlr->cp_val.volume, idx);
+	vcs_client_peer[idx].master_vol = vol_ctlr->cp_val.volume;
+	vcs_client_peer[idx].unknown_vol = false;
+}
+
+static void vcs_state_cb_handler(struct bt_vcp_vol_ctlr *vcs, int err, uint8_t dev_volume,
+				 uint8_t mute)
+{
+	int ret;
+	int current_idx = vcs_peer_index_get(vcs);
+
 	if (err) {
 		LOG_ERR("VCS state callback error: %d", err);
 		return;
 	}
 
-	for (int i = 0; i < ARRAY_SIZE(vcs_client_peer); i++) {
-		if (vcs == vcs_client_peer[i]) {
-			LOG_DBG("VCS state from remote device %d:", i);
-			continue;
-		}
+	if (current_idx < 0) {
+		LOG_ERR("VCS peer not found");
+		return;
+	}
 
-		LOG_DBG("Sync with other devices %d", i);
+	LOG_INF("VCS state ctrl cb idx: %u vol: %u mute: %u", current_idx, dev_volume, mute);
 
-		if (vcs_client_peer[i] == NULL) {
-			/* Skip */
-			continue;
+	if (vcs_client_peer[current_idx].unknown_vol) {
+		LOG_DBG("VCS default state from device: %u", current_idx);
+		uint8_t common_vol;
+
+		ret = common_vol_get();
+		if (ret < 0) {
+			LOG_DBG("Saving vol %u for renderer %u", dev_volume, current_idx);
+			vcs_client_peer[current_idx].master_vol = dev_volume;
+			vcs_client_peer[current_idx].unknown_vol = false;
+			return;
 		}
 
-		ret = bt_vcp_vol_ctlr_set_vol(vcs_client_peer[i], volume);
-		if (ret) {
-			LOG_DBG("Failed to sync volume to remote device %d, err = "
-				"%d",
-				i, ret);
+		common_vol = ret;
+
+		if (common_vol == dev_volume) {
+			/* Volume of this device is the same as the common volume. No update */
+			return;
 		}
+
+		LOG_DBG("Stored vol found. Writing vol %u to remote device %u", common_vol,
+			current_idx);
+		update_volume_of_all_peers(vcs, common_vol, true);
+
+	} else {
+		/* Write vol change to all other devices. No need to update caller */
+		update_volume_of_all_peers(vcs, dev_volume, false);
 	}
 }
 
-/**
- * @brief	Callback handler for the VCS controller flags.
- *
- * @note	This callback handler will be triggered if VCS flags changed.
- */
-static void vcs_flags_ctlr_cb_handler(struct bt_vcp_vol_ctlr *vcs, int err, uint8_t flags)
```

Comment:
Why was the documentation removed?

Reply:
These are static functions and their documentation should reside where this callback is defined in the host.